experimental/Schemes/ToricSchemes/NormalToricSchemes/Attributes.jl

```diff
@@ -18,7 +18,112 @@ julia> underlying_scheme(toric_scheme)
 covered scheme with 3 affine patches in its default covering
 ```
 """
-underlying_scheme(X::ToricCoveredScheme) = X.X
+function underlying_scheme(Z::ToricCoveredScheme)
+  if !isdefined(Z, :X)
+    F = fan(normal_toric_variety(Z))
+    C = maximal_cones(F)
+
+    antv = affine_open_covering(normal_toric_variety(Z))
+    var_names = [Symbol.(["x_$(i)_$(k)" for i in 1:ngens(base_ring(toric_ideal(antv[k])))]) for k in 1:length(antv)]
+    #rings = [polynomial_ring(coefficient_ring(antv[k]), ["x_$(i)_$(k)" for i in 1:ngens(base_ring(toric_ideal(antv[k])))], cached=false)[1] for k in 1:length(antv)]
+    patch_list = [ToricSpec(U, var_names=n) for (U, n) in zip(antv, var_names)]
+    cov = Covering(patch_list)
+
+    for i in 1:(length(patch_list)-1)
+      X = patch_list[i]
+      CX = cone(X)
+      CXdual = dual_cone(X)
+      for j in i+1:length(patch_list)
+        Y = patch_list[j]
+        CY = cone(Y)
+        CYdual = dual_cone(Y)
+
+        # Now glue X and Y
+        
+        # Step 1: Find the cone of the intersection of X and Y.
+        facet = intersect(CX, CY)
+        # harvest the dual cone
+        dual_facet = polarize(facet)
+        if !(dim(facet) == dim(CX) - 1)
+          continue
+        end
+
+        # Step 2: Extract the candidates for the localization element.
+        candidates = dual_facet.pm_cone.HILBERT_BASIS_GENERATORS[2]
+        v = candidates[1,:]
+        for j in 1:nrows(candidates)
+          v = candidates[j,:]
+          if contains(CXdual, v) && contains(CYdual, -v)
+            break
+          elseif contains(CXdual, -v) && contains(CYdual, v)
+            v = -v
+            break
+          end
+        end
+        (contains(CXdual, v) && contains(CYdual, -v)) || error("no element found for localization")
+        
+        # Step 3: We express the loclization element (it is v, resp. -v) in the Hilbert basis of the two patches.
+        # -> Then localize at this element.
+        vmat = matrix(ZZ.(collect(v)))
+        
+        AX = transpose(hilbert_basis(X))
+        Id = identity_matrix(ZZ, ncols(AX))
+        sol = solve_mixed(AX, vmat, Id, zero(vmat))
+        fX = prod([x^k for (x, k) in zip(gens(ambient_coordinate_ring(X)), sol)])
+        U = PrincipalOpenSubset(X, OO(X)(fX))
+        
+        AY = transpose(hilbert_basis(Y))
+        Id = identity_matrix(ZZ, ncols(AY))
+        sol = solve_mixed(AY, -vmat, Id, zero(vmat))
+        fY = prod([x^(k>0 ? 1 : 0) for (x, k) in zip(gens(ambient_coordinate_ring(Y)), sol)])
```

Comment:
why are `fY` and `fX` computed differently? I would have expected it to be symmetric.

Reply:
It probably isn't symmetric because the joint faces of the two cones and their lineality spaces have a relative orientation with the two cones. Then we start with the one for `X` choosing such orientations, so that no negative signs can occur there. For `Y` it seems that we can run into problems, then. But this is just a gut-feeling right now. I would have to look through it again. 

The thing is: It's the same code as before and it has already been reviewed once. So I would just keep it as is for now.